# libhubbub: treebuilder
# 4 files, 228 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libhubbub/src/treebuilder/autogenerated-element-type.c browser/libhubbub/src/treebuilder/autogenerated-element-type.c
--- upstream/libhubbub/src/treebuilder/autogenerated-element-type.c	2026-09-20 23:27:37.382598331 +0100
+++ browser/libhubbub/src/treebuilder/autogenerated-element-type.c	2026-09-20 23:20:29.353872079 +0100
@@ -1,5 +1,5 @@
 /* ANSI-C code produced by gperf version 3.1 */
-/* Command-line: gperf --output-file=src/treebuilder/autogenerated-element-type.c.tmp src/treebuilder/element-type.gperf  */
+/* Command-line: gperf --output-file=browser/libhubbub/src/treebuilder/autogenerated-element-type.c.tmp browser/libhubbub/src/treebuilder/element-type.gperf  */
 /* Computed positions: -k'1-2,$' */
 
 #if !((' ' == 32) && ('!' == 33) && ('"' == 34) && ('#' == 35) \
@@ -29,13 +29,13 @@
 #error "gperf generated tables don't work with this execution character set. Please report a bug to <bug-gperf@gnu.org>."
 #endif
 
-#line 18 "src/treebuilder/element-type.gperf"
+#line 18 "browser/libhubbub/src/treebuilder/element-type.gperf"
 
 #include <string.h>
 
 #include "treebuilder/element-type.h"
 
-#line 25 "src/treebuilder/element-type.gperf"
+#line 25 "browser/libhubbub/src/treebuilder/element-type.gperf"
 struct element_type_map;
 
 #define TOTAL_KEYWORDS 106
@@ -145,217 +145,217 @@
 
 static const struct element_type_map wordlist[] =
   {
-#line 27 "src/treebuilder/element-type.gperf"
+#line 27 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"a",                   A},
-#line 127 "src/treebuilder/element-type.gperf"
+#line 127 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"tr",                  TR},
-#line 31 "src/treebuilder/element-type.gperf"
+#line 31 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"area",                AREA},
-#line 30 "src/treebuilder/element-type.gperf"
+#line 30 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"applet",              APPLET},
-#line 61 "src/treebuilder/element-type.gperf"
+#line 61 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"font",                FONT},
-#line 62 "src/treebuilder/element-type.gperf"
+#line 62 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"footer",              FOOTER},
-#line 66 "src/treebuilder/element-type.gperf"
+#line 66 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"frameset",            FRAMESET},
-#line 64 "src/treebuilder/element-type.gperf"
+#line 64 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"form",                FORM},
-#line 113 "src/treebuilder/element-type.gperf"
+#line 113 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"spacer",              SPACER},
-#line 32 "src/treebuilder/element-type.gperf"
+#line 32 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"article",             ARTICLE},
-#line 63 "src/treebuilder/element-type.gperf"
+#line 63 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"foreignobject",       FOREIGNOBJECT},
-#line 96 "src/treebuilder/element-type.gperf"
+#line 96 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"nobr",                NOBR},
-#line 65 "src/treebuilder/element-type.gperf"
+#line 65 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"frame",               FRAME},
-#line 109 "src/treebuilder/element-type.gperf"
+#line 109 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"s",                   S},
-#line 28 "src/treebuilder/element-type.gperf"
+#line 28 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"address",             ADDRESS},
-#line 99 "src/treebuilder/element-type.gperf"
+#line 99 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"noscript",            NOSCRIPT},
-#line 110 "src/treebuilder/element-type.gperf"
+#line 110 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"script",              SCRIPT},
-#line 121 "src/treebuilder/element-type.gperf"
+#line 121 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"td",                  TD},
-#line 58 "src/treebuilder/element-type.gperf"
+#line 58 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"fieldset",            FIELDSET},
-#line 126 "src/treebuilder/element-type.gperf"
+#line 126 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"title",               TITLE},
-#line 98 "src/treebuilder/element-type.gperf"
+#line 98 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"noframes",            NOFRAMES},
-#line 60 "src/treebuilder/element-type.gperf"
+#line 60 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"figure",              FIGURE},
-#line 74 "src/treebuilder/element-type.gperf"
+#line 74 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"hr",                  HR},
-#line 52 "src/treebuilder/element-type.gperf"
-    {"dir",                 DIR},
-#line 123 "src/treebuilder/element-type.gperf"
+#line 52 "browser/libhubbub/src/treebuilder/element-type.gperf"
+    {"dir",                 HUBBUB_DIR},
+#line 123 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"tfoot",               TFOOT},
-#line 97 "src/treebuilder/element-type.gperf"
+#line 97 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"noembed",             NOEMBED},
-#line 122 "src/treebuilder/element-type.gperf"
+#line 122 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"textarea",            TEXTAREA},
-#line 89 "src/treebuilder/element-type.gperf"
+#line 89 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"meta",                META},
-#line 59 "src/treebuilder/element-type.gperf"
+#line 59 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"figcaption",          FIGCAPTION},
-#line 48 "src/treebuilder/element-type.gperf"
+#line 48 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"dd",                  DD},
-#line 108 "src/treebuilder/element-type.gperf"
+#line 108 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"pre",                 PRE},
-#line 111 "src/treebuilder/element-type.gperf"
+#line 111 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"select",              SELECT},
-#line 72 "src/treebuilder/element-type.gperf"
+#line 72 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h6",                  H6},
-#line 119 "src/treebuilder/element-type.gperf"
+#line 119 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"table",               TABLE},
-#line 92 "src/treebuilder/element-type.gperf"
+#line 92 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"mn",                  MN},
-#line 118 "src/treebuilder/element-type.gperf"
+#line 118 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"svg",                 SVG},
-#line 107 "src/treebuilder/element-type.gperf"
+#line 107 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"plaintext",           PLAINTEXT},
-#line 86 "src/treebuilder/element-type.gperf"
+#line 86 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"marquee",             MARQUEE},
-#line 131 "src/treebuilder/element-type.gperf"
+#line 131 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"wbr",                 WBR},
-#line 47 "src/treebuilder/element-type.gperf"
+#line 47 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"command",             COMMAND},
-#line 85 "src/treebuilder/element-type.gperf"
+#line 85 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"malignmark",          MALIGNMARK},
-#line 105 "src/treebuilder/element-type.gperf"
+#line 105 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"p",                   P},
-#line 50 "src/treebuilder/element-type.gperf"
+#line 50 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"details",             DETAILS},
-#line 51 "src/treebuilder/element-type.gperf"
+#line 51 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"dialog",              DIALOG},
-#line 93 "src/treebuilder/element-type.gperf"
+#line 93 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"mo",                  MO},
-#line 44 "src/treebuilder/element-type.gperf"
+#line 44 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"center",              CENTER},
-#line 91 "src/treebuilder/element-type.gperf"
+#line 91 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"mi",                  MI},
-#line 46 "src/treebuilder/element-type.gperf"
+#line 46 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"colgroup",            COLGROUP},
-#line 87 "src/treebuilder/element-type.gperf"
+#line 87 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"math",                MATH},
-#line 106 "src/treebuilder/element-type.gperf"
+#line 106 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"param",               PARAM},
-#line 56 "src/treebuilder/element-type.gperf"
+#line 56 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"em",                  EM},
-#line 83 "src/treebuilder/element-type.gperf"
+#line 83 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"link",                LINK},
-#line 120 "src/treebuilder/element-type.gperf"
+#line 120 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"tbody",               TBODY},
-#line 103 "src/treebuilder/element-type.gperf"
+#line 103 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"option",              OPTION},
-#line 54 "src/treebuilder/element-type.gperf"
+#line 54 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"dl",                  DL},
-#line 73 "src/treebuilder/element-type.gperf"
+#line 73 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"head",                HEAD},
-#line 125 "src/treebuilder/element-type.gperf"
+#line 125 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"thead",               THEAD},
-#line 100 "src/treebuilder/element-type.gperf"
+#line 100 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"object",              OBJECT},
-#line 41 "src/treebuilder/element-type.gperf"
+#line 41 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"br",                  BR},
-#line 45 "src/treebuilder/element-type.gperf"
+#line 45 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"col",                 COL},
-#line 49 "src/treebuilder/element-type.gperf"
+#line 49 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"desc",                DESC},
-#line 80 "src/treebuilder/element-type.gperf"
+#line 80 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"input",               INPUT},
-#line 43 "src/treebuilder/element-type.gperf"
+#line 43 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"caption",             CAPTION},
-#line 29 "src/treebuilder/element-type.gperf"
+#line 29 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"annotation-xml",      ANNOTATION_XML},
-#line 57 "src/treebuilder/element-type.gperf"
+#line 57 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"embed",               EMBED},
-#line 90 "src/treebuilder/element-type.gperf"
+#line 90 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"mglyph",              MGLYPH},
-#line 124 "src/treebuilder/element-type.gperf"
+#line 124 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"th",                  TH},
-#line 77 "src/treebuilder/element-type.gperf"
+#line 77 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"iframe",              IFRAME},
-#line 84 "src/treebuilder/element-type.gperf"
+#line 84 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"listing",             LISTING},
-#line 102 "src/treebuilder/element-type.gperf"
+#line 102 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"optgroup",            OPTGROUP},
-#line 33 "src/treebuilder/element-type.gperf"
+#line 33 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"aside",               ASIDE},
-#line 104 "src/treebuilder/element-type.gperf"
+#line 104 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"output",              OUTPUT},
-#line 94 "src/treebuilder/element-type.gperf"
+#line 94 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"ms",                  MS},
-#line 132 "src/treebuilder/element-type.gperf"
+#line 132 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"xmp",                 XMP},
-#line 76 "src/treebuilder/element-type.gperf"
+#line 76 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"i",                   I},
-#line 117 "src/treebuilder/element-type.gperf"
+#line 117 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"summary",             SUMMARY},
-#line 128 "src/treebuilder/element-type.gperf"
+#line 128 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"tt",                  TT},
-#line 39 "src/treebuilder/element-type.gperf"
+#line 39 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"blockquote",          BLOCKQUOTE},
-#line 82 "src/treebuilder/element-type.gperf"
+#line 82 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"li",                  LI},
-#line 95 "src/treebuilder/element-type.gperf"
+#line 95 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"mtext",               MTEXT},
-#line 71 "src/treebuilder/element-type.gperf"
+#line 71 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h5",                  H5},
-#line 112 "src/treebuilder/element-type.gperf"
+#line 112 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"small",               SMALL},
-#line 101 "src/treebuilder/element-type.gperf"
+#line 101 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"ol",                  OL},
-#line 36 "src/treebuilder/element-type.gperf"
+#line 36 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"basefont",            BASEFONT},
-#line 88 "src/treebuilder/element-type.gperf"
+#line 88 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"menu",                MENU},
-#line 78 "src/treebuilder/element-type.gperf"
+#line 78 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"image",               IMAGE},
-#line 55 "src/treebuilder/element-type.gperf"
+#line 55 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"dt",                  DT},
-#line 38 "src/treebuilder/element-type.gperf"
+#line 38 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"big",                 BIG},
-#line 35 "src/treebuilder/element-type.gperf"
+#line 35 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"base",                BASE},
-#line 116 "src/treebuilder/element-type.gperf"
+#line 116 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"style",               STYLE},
-#line 114 "src/treebuilder/element-type.gperf"
+#line 114 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"strike",              STRIKE},
-#line 70 "src/treebuilder/element-type.gperf"
+#line 70 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h4",                  H4},
-#line 42 "src/treebuilder/element-type.gperf"
+#line 42 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"button",              BUTTON},
-#line 69 "src/treebuilder/element-type.gperf"
+#line 69 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h3",                  H3},
-#line 68 "src/treebuilder/element-type.gperf"
+#line 68 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h2",                  H2},
-#line 53 "src/treebuilder/element-type.gperf"
+#line 53 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"div",                 DIV},
-#line 40 "src/treebuilder/element-type.gperf"
+#line 40 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"body",                BODY},
-#line 67 "src/treebuilder/element-type.gperf"
+#line 67 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"h1",                  H1},
-#line 79 "src/treebuilder/element-type.gperf"
+#line 79 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"img",                 IMG},
-#line 130 "src/treebuilder/element-type.gperf"
+#line 130 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"ul",                  UL},
-#line 115 "src/treebuilder/element-type.gperf"
+#line 115 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"strong",              STRONG},
-#line 81 "src/treebuilder/element-type.gperf"
+#line 81 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"isindex",             ISINDEX},
-#line 37 "src/treebuilder/element-type.gperf"
+#line 37 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"bgsound",             BGSOUND},
-#line 34 "src/treebuilder/element-type.gperf"
+#line 34 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"b",                   B},
-#line 129 "src/treebuilder/element-type.gperf"
+#line 129 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"u",                   U},
-#line 75 "src/treebuilder/element-type.gperf"
+#line 75 "browser/libhubbub/src/treebuilder/element-type.gperf"
     {"html",                HTML}
   };
 
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libhubbub/src/treebuilder/element-type.gperf browser/libhubbub/src/treebuilder/element-type.gperf
--- upstream/libhubbub/src/treebuilder/element-type.gperf	2026-09-20 23:27:11.702644814 +0100
+++ browser/libhubbub/src/treebuilder/element-type.gperf	2026-09-20 23:19:17.815883858 +0100
@@ -49,7 +49,7 @@
 desc,                DESC
 details,             DETAILS
 dialog,              DIALOG
-dir,                 DIR
+dir,                 HUBBUB_DIR
 div,                 DIV
 dl,                  DL
 dt,                  DT
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libhubbub/src/treebuilder/element-type.h browser/libhubbub/src/treebuilder/element-type.h
--- upstream/libhubbub/src/treebuilder/element-type.h	2026-09-20 23:27:11.702644814 +0100
+++ browser/libhubbub/src/treebuilder/element-type.h	2026-09-20 23:19:17.815998339 +0100
@@ -16,7 +16,7 @@
 /* Special */
 	ADDRESS, AREA, ARTICLE, ASIDE, BASE, BASEFONT, BGSOUND, BLOCKQUOTE,
 	BODY, BR, CENTER, COL, COLGROUP, COMMAND, DATAGRID, DD, DETAILS,
-	DIALOG, DIR, DIV, DL, DT, EMBED, FIELDSET, FIGCAPTION, FIGURE, FOOTER,
+	DIALOG, HUBBUB_DIR, DIV, DL, DT, EMBED, FIELDSET, FIGCAPTION, FIGURE, FOOTER,
 	FORM, FRAME, FRAMESET, H1, H2, H3, H4, H5, H6, HEAD, HEADER, HR, IFRAME,
 	IMAGE, IMG, INPUT, ISINDEX, LI, LINK, LISTING, MAIN, MENU, META, NAV,
 	NOEMBED, NOFRAMES, NOSCRIPT, OL, OPTGROUP, OPTION, P, PARAM, PLAINTEXT,
diff -urN -x .git -x 'build-*' -x test -x '*.o' -x '*.a' -x '*.d' -x '*.pc' upstream/libhubbub/src/treebuilder/in_body.c browser/libhubbub/src/treebuilder/in_body.c
--- upstream/libhubbub/src/treebuilder/in_body.c	2026-09-20 23:27:11.702644814 +0100
+++ browser/libhubbub/src/treebuilder/in_body.c	2026-09-20 23:19:17.816289410 +0100
@@ -265,7 +265,7 @@
 	} else if (type == ADDRESS || type == ARTICLE || type == ASIDE ||
 			type == BLOCKQUOTE || type == CENTER ||
 			type == DATAGRID || type == DETAILS ||
-			type == DIALOG || type == DIR ||
+			type == DIALOG || type == HUBBUB_DIR ||
 			type == DIV || type == DL || type == FIELDSET ||
 			type == FIGCAPTION || type == FIGURE ||
 			type == FOOTER || type == HEADER || type == MAIN ||
@@ -432,7 +432,7 @@
 		err = process_0body_in_body(treebuilder);
 	} else if (type == ADDRESS || type == ARTICLE || type == ASIDE ||
 			type == BLOCKQUOTE || type == CENTER ||
-			type == DETAILS || type == DIALOG || type == DIR ||
+			type == DETAILS || type == DIALOG || type == HUBBUB_DIR ||
 			type == DATAGRID || type == DIV || type == DL ||
 			type == FIELDSET || type == FIGCAPTION ||
 			type == FIGURE || type == FOOTER || type == HEADER ||
